org-roam.el

```diff
@@ -619,6 +638,53 @@ If item at point is not org-roam specific, default to Org behaviour."
              (select-window org-roam-last-window))
     (find-file file)))
 
+(defun org-roam--unlinked-regexp (file-path buffer-title)
+  (rx-to-string `(: word-boundary
+                    (or ,buffer-title
+                        ,(downcase buffer-title)
+                        ,(file-name-base file-path))
+                    word-boundary)))
+
+(defun org-roam--find-unlinked-references (file-path buffer-title files)
+  (let* ((match-regexp (org-roam--unlinked-regexp file-path buffer-title))
+         (find-func (lambda (other-file)
+                      (when (not (string-equal other-file file-path))
+                        (with-temp-buffer
+                          (insert-file-contents other-file)
+                          (let ((matches)
+                                (unwanted-matches))
+                            (dolist (unwanted-regexp org-roam-unwanted-regexps)
+                              (while (re-search-forward unwanted-regexp nil t)
+                                (push (list :start (match-beginning 0) :end (match-end 0)) unwanted-matches))
+                              (goto-char (point-min)))
+                            (org-mode)
+                            (org-toggle-link-display)
```

Comment:
Are these really needed?

Reply:
The issue is the location of the point. If you have any links in the file, `with-temp-buffer insert-file-contents` does not collapse them. This means the point is relative to the **raw** file. However, users see the formatted file, so the point will point in a different place, e.g:

```
[[linkhere][text]]

(point)unlinked here

some text after
```

```
_text_

unlinked here

som(point)e text after
```

 Is there a way to get the location of the text in another way?